api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxPvlsBySourceQueries.java

```diff
@@ -0,0 +1,257 @@
+      int adultSeguimentoEncounter,
+      int pediatriaSeguimentoEncounter,
+      int mastercardEncounter,
+      int vlConceptQuestion,
+      int vlQualitativeConceptQuestion) {
+    Map<String, String> map = new HashMap<>();
+    map.put("6", String.valueOf(adultSeguimentoEncounter));
+    map.put("9", String.valueOf(pediatriaSeguimentoEncounter));
+    map.put("53", String.valueOf(mastercardEncounter));
+    map.put("856", String.valueOf(vlConceptQuestion));
+    map.put("1305", String.valueOf(vlQualitativeConceptQuestion));
+
+    String query =
+        "SELECT p.patient_id FROM  patient p INNER JOIN encounter e ON p.patient_id=e.patient_id INNER JOIN"
+            + " obs o ON e.encounter_id=o.encounter_id "
+            + " WHERE p.voided=0 AND e.voided=0 AND o.voided=0 AND "
+            + " e.encounter_type IN (${6},${9}) AND "
+            + " ((o.concept_id=${856} AND o.value_numeric IS NOT NULL) OR (o.concept_id=${1305} AND o.value_coded IS NOT NULL)) AND "
+            + " e.encounter_datetime BETWEEN date_add(date_add(:endDate, interval -12 MONTH), interval 1 day) AND :endDate AND "
+            + " e.location_id=:location "
+            + " UNION "
+            + " SELECT p.patient_id FROM  patient p INNER JOIN encounter e ON p.patient_id=e.patient_id INNER JOIN "
+            + " obs o ON e.encounter_id=o.encounter_id "
+            + " WHERE p.voided=0 AND e.voided=0 AND o.voided=0 AND "
+            + " e.encounter_type IN (${53}) AND o.concept_id=${856} AND o.value_numeric IS NOT NULL AND "
+            + " o.obs_datetime BETWEEN date_add(date_add(:endDate, interval -12 MONTH), interval 1 day) AND :endDate AND "
+            + " e.location_id=:location ";
+    StringSubstitutor sb = new StringSubstitutor(map);
+    return sb.replace(query);
+  }
+
+  /**
+   * <b>Description</b>Patients with viral load suppression within 12 months
+   *
+   * @param adultSeguimentoEncounter
+   * @param pediatriaSeguimentoEncounter
+   * @param mastercardEncounter
+   * @param vlConceptQuestion
+   * @param vlQualitativeConceptQuestion
+   * @return String
+   */
+  public static String getPatientsWithViralLoadSuppressionPvlsFichaMastreNumerator(
+      int adultSeguimentoEncounter,
+      int pediatriaSeguimentoEncounter,
+      int mastercardEncounter,
+      int vlConceptQuestion,
+      int vlQualitativeConceptQuestion) {
+
+    Map<String, String> map = new HashMap<>();
+    map.put("6", String.valueOf(adultSeguimentoEncounter));
+    map.put("9", String.valueOf(pediatriaSeguimentoEncounter));
+    map.put("53", String.valueOf(mastercardEncounter));
+    map.put("856", String.valueOf(vlConceptQuestion));
+    map.put("1305", String.valueOf(vlQualitativeConceptQuestion));
+    String query =
+        "SELECT fn1.patient_id FROM( "
+            + " SELECT  patient_id, MAX(data_carga) AS data_carga FROM( "
+            + " SELECT patient_id,data_carga FROM "
+            + " (SELECT p.patient_id,MAX(o.obs_datetime) data_carga "
```

Comment:
Please check against the spec if should be encounter_datetime or obs_datetime.

Reply:
@ningosi can you please check this, I believe this should be encounter_datetime